src/Bass/Shared/ChannelReferences.cs

```diff
@@ -9,17 +11,25 @@ namespace ManagedBass
     /// </summary>
     public static class ChannelReferences
     {
-#if !__IOS__
         static readonly ConcurrentDictionary<Tuple<int, int>, object> Procedures = new ConcurrentDictionary<Tuple<int, int>, object>();
         static readonly SyncProcedure Freeproc = Callback;
-#endif
+
+        private static bool JITSupported;
+
+        static ChannelReferences()
+        {
+            var assemblies = AppDomain.CurrentDomain.GetAssemblies().Select(Asm => Asm.ToString());
```

Comment:
Surely this is quite a commonly required check in other projects (ie. official .net internals) – is there an example of it being done this way elsewhere?

Reply:
Seems like you may be able to use https://learn.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.runtimefeature.isdynamiccodesupported?view=net-7.0#system-runtime-compilerservices-runtimefeature-isdynamiccodesupported

Reply:
Nice one. That only works for .NET iOS so will bring back the constants usage for Xamarin's case (in case any Xamarin project rely on `ManagedBass`).